contracts/utils/types/Time.sol

```diff
@@ -103,10 +103,15 @@ library Time {
      * enforce the old delay at the moment of the update. Returns the updated Delay object and the timestamp when the
      * new delay becomes effective.
      */
-    function withUpdate(Delay self, uint32 newValue, uint32 minSetback) internal view returns (Delay, uint48) {
-        uint32 value = self.get();
+    function withUpdate(
+        Delay self,
+        uint48 clock,
+        uint32 newValue,
+        uint32 minSetback
+    ) internal pure returns (Delay, uint48) {
```

Comment:
The issue with this function is that `clock` is expected to always be the "current" timepoint, and the semantics of passing some other timepoint are quite unclear: if there is some existing pending delay update, `withUpdate` does not result in a delay object with two future updates, it may just jump ahead of time to the existing future update.